### ISSUE TITLE
Ability to serve static HTML

### ISSUE THREAD

**Issue by TehShrike** [TehShrike/noddity#30]
We need the ability to serve up static HTML that is valid (working links, asset references) content.

For a while I was leaning towards an http server that would generate the html on the fly.  But at this point I'm leaning towards a traditional static file generator just for simplicity's sake.

Generating HTML files that look the same wouldn't be too hard.  Difficulties may include:
- relative links work (including links to pages in subfolders of the content folder)
- images should display correctly (even when in a subfolder of the content folder)
- the static HTML should be replaced by the live Ractive version (without any visible blinking) once the content is loaded (incidentally, should be much more efficient after https://github.com/ractivejs/ractive/pull/2196 )
- ideally the builder will be able to work only on files that have been altered since the last build.  But changing a template should also cause all the posts that use it to also be updated.  Somehow.

Anything else?

**Comment by spacetag** [comment]
Is the goal to let non-javascripted visitors (people on old mobile phones, some search engines) view Noddity content? If so, cool!

**Comment by TehShrike** [comment]
Yeah, non-JavaScript browsers are the main driver.  Search engines are already covered by seoaas.
